service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java

```diff
@@ -0,0 +1,130 @@
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.exception.EntityAlreadyExistsException;
+import com.epam.esm.exception.EntityNotFoundException;
+import com.epam.esm.exception.ValidatorException;
+import com.epam.esm.repository.CertificateRepository;
+import com.epam.esm.repository.TagRepository;
+import com.epam.esm.service.CertificateService;
+import com.epam.esm.service.util.CertificateParamsRequestUtil;
+import com.epam.esm.specification.CertificateAllSpecification;
+import com.epam.esm.specification.SqlSpecification;
+import com.epam.esm.validator.CertificateDTOValidator;
+import com.epam.esm.validator.CertificatePageQueryDTOValidator;
+import com.epam.esm.validator.Validator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class CertificateServiceImpl implements CertificateService {
+    private final CertificateRepository certificateRepository;
+    private final TagRepository tagRepository;
+    private final Validator<CertificateDTO> certificateDTOValidator;
+    private final Validator<CertificatePageQueryDTO> pageQueryDTOValidator;
+
+    @Autowired
+    public CertificateServiceImpl(CertificateRepository certificateRepository, TagRepository tagRepository, CertificateDTOValidator certificateDTOValidator, CertificatePageQueryDTOValidator pageQueryDTOValidator) {
+        this.certificateRepository = certificateRepository;
+        this.tagRepository = tagRepository;
+        this.certificateDTOValidator = certificateDTOValidator;
+        this.pageQueryDTOValidator = pageQueryDTOValidator;
+    }
+
+    @Override
+    public CertificateDTO getById(Long id) {
+        Certificate certificate = certificateRepository.getById(id).orElseThrow(() -> new EntityNotFoundException(" (id = " + id + ")"));
+        addTagsToCertificate(certificate);
+        return CertificateConverterDTO.convertToDto(certificate);
+    }
+
+
+    @Override
+    @Transactional
+    public CertificateDTO create(CertificateDTO certificateDTO) {
+        certificateDTOValidator.cleanErrorMessages();
+        if (!certificateDTOValidator.isValid(certificateDTO)) {
+            throw new ValidatorException(certificateDTOValidator.getErrorMessage());
+        }
+        String name = certificateDTO.getName();
+        if (certificateRepository.getByName(name).isPresent()) {
+            throw new EntityAlreadyExistsException(" (name = " + name + ")");
+        }
+        Certificate certificate = CertificateConverterDTO.convertToEntity(certificateDTO);
+        checkTags(certificate);
+        certificate = certificateRepository.save(certificate);
+        tagRepository.createCertificateTags(certificate);
+        certificate = certificateRepository.getById(certificate.getId()).orElseThrow(EntityNotFoundException::new);
```

Comment:
in the same code (line 44) you pass "id" to create new EntityNotFoundException. why here without "id"?